src/ContractInteract/EIP20Gateway.js

```diff
@@ -68,6 +97,244 @@ class EIP20Gateway {
+   * Setup for EIP20Gateway. Deploys EIP20Gateway and EIP20CoGateway,
+   * and links them.
+   *
+   * @param {Object} originWeb3 Web3 object.
+   * @param {Object} auxiliaryWeb3 Web3 object.
+   * @param {EIP20GatewaySetupConfig} gatewayConfig EIP20Gateway setup configuration.
+   * @param {EIP20CoGatewaySetupConfig} coGatewayConfig EIP20CoGateway setup configuration.
+   * @param {Object} originTxOptions Transaction options on Origin.
+   * @param {Object} auxiliaryTxOptions Transaction options on Auxiliary.
+   *
+   * @returns {Promise<EIP20Gateway>} Promise containing the EIP20Gateway instance that
+   *                                  has been set up.
+   */
+  static setup(
+    originWeb3,
+    auxiliaryWeb3,
+    gatewayConfig,
+    coGatewayConfig,
+    originTxOptions = {},
+    auxiliaryTxOptions = {},
+  ) {
+    if (!gatewayConfig) {
+      throw new Error('Mandatory parameter "gatewayConfig" missing. ');
+    }
+    if (!coGatewayConfig) {
+      throw new Error('Mandatory parameter "coGatewayConfig" missing. ');
+    }
+
+    gatewayConfig = gatewayConfig || {};
+    gatewayConfig.messageBus = gatewayConfig.messageBus || this.messageBus;
+    gatewayConfig.gatewayLib = gatewayConfig.gatewayLib || this.gatewayLib;
+
+    if (!originWeb3) {
+      throw new Error('Mandatory parameter "originWeb3" missing.');
+    }
+
+    if (!auxiliaryWeb3) {
+      throw new Error('Mandatory parameter "auxiliaryWeb3" missing.');
+    }
+
+    if (!gatewayConfig.baseToken) {
+      throw new Error(
+        'Mandatory configuration "baseToken" missing. Set gatewayConfig.baseToken address',
+      );
+    }
+
+    EIP20Gateway.validateSetupConfig(gatewayConfig);
+    EIP20CoGateway.validateSetupConfig(coGatewayConfig);
+
+    const gatewayDeployParams = Object.assign({}, originTxOptions);
+    gatewayDeployParams.from = gatewayConfig.deployer;
+
+    const gatewayDeploy = EIP20Gateway.deploy(
+      originWeb3,
+      gatewayConfig.token,
+      gatewayConfig.baseToken,
+      gatewayConfig.anchor,
+      gatewayConfig.bounty,
+      gatewayConfig.organization,
+      gatewayConfig.burner,
+      gatewayConfig.messageBus,
+      gatewayConfig.gatewayLib,
+      gatewayDeployParams,
+    );
+
+    const coGatewayDeploy = gatewayDeploy.then((gateway) => {
+      const gatewayAddress = gateway.address;
+      coGatewayConfig.gateway = gatewayAddress;
+
+      return EIP20CoGateway.setup(
+        auxiliaryWeb3,
+        coGatewayConfig,
+        auxiliaryTxOptions,
+      );
+    });
+
+    const gatewayActivation = Promise.all([
+      gatewayDeploy,
+      coGatewayDeploy,
+    ]).then(([gateway, coGateway]) => {
+      const ownerTxParams = Object.assign({}, gatewayDeployParams);
+      ownerTxParams.from = gatewayConfig.organizationOwner;
+
+      const coGatewayAddress = coGateway.address;
+
+      return gateway
+        .activateGateway(coGatewayAddress, ownerTxParams)
+        .then(() => gateway);
+    });
+
+    return gatewayActivation;
```

Comment:
Should this not return the gateway and the co-gateway?

Reply:
Hmm, I intended it to just return the gateway for consistency with the other `setup` methods. You could then retrieve the co-gateway from the the gateway (though the necessary methods don't seem to be implemented yet).

Reply:
rename to `setupPair` and return both contracts